syncapi/routing/routing.go

```diff
@@ -42,6 +42,7 @@ func Setup(
 	r0mux := csMux.PathPrefix("/r0").Subrouter()
 
 	// TODO: Add AS support for all handlers below.
+	// asdf
 	r0mux.Handle("/sync", httputil.MakeAuthAPI("sync", userAPI, func(req *http.Request, device *userapi.Device) util.JSONResponse {
```

Comment:
I am trying to get Sytest to run for the `Archived rooms only contain history from before the user left` test but it just says `SKIP due to lack of can_sync`. I am not sure where `can_sync` is defined and how to fix this?


```sh
$ ./run-tests.pl -I Dendrite::Monolith -W ../dendrite/sytest-whitelist
...
Running tests/31sync/09archived.pl...
  Testing if: Left rooms appear in the leave section of sync... SKIP due to lack of can_sync
  Testing if: Newly left rooms appear in the leave section of incremental sync... SKIP due to lack of can_sync
  Testing if: We should see our own leave event, even if history_visibility is restricted (SYN-662)... SKIP due to lack of can_sync
  Testing if: We should see our own leave event when rejecting an invite, even if history_visibility is restricted (riot-web/3462)... SKIP due to lack of can_sync
  Testing if: Newly left rooms appear in the leave section of gapped sync... SKIP due to lack of can_sync
  Testing if: Previously left rooms don't appear in the leave section of sync... SKIP due to lack of can_sync
  Testing if: Left rooms appear in the leave section of full state sync... SKIP due to lack of can_sync
  Testing if: Archived rooms only contain history from before the user left... SKIP due to lack of can_sync
```

Reply:
My best guess so far:

 - [tests/31sync/02sync.pl#L7](https://github.com/matrix-org/sytest/blob/5a63b76ba5d79a9cf7fc5827ce3d250b0846b834/tests/31sync/02sync.pl#L7) which does `proves => [qw( can_sync )],`
 - The `matrix_sync` definition is in [`tests/10apidoc/03events-initial.pl#L208-L237`](https://github.com/matrix-org/sytest/blob/5a63b76ba5d79a9cf7fc5827ce3d250b0846b834/tests/10apidoc/03events-initial.pl#L208-L237)